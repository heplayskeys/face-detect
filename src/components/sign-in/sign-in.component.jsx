import React, { useState } from 'react';
import { connect } from 'react-redux';
import { updateRoute } from '../../redux/route/route.actions';
import { setUserProfile } from '../../redux/user/user.actions';
import './sign-in.styles.scss';

export const SignIn = ({ URL, route, setRoute, setUser }) => {
	const [errorState, setErrorState] = useState('');

	const handleSubmit = async () => {
		const email = document.querySelector('#email-address').value;
		const password = document.querySelector('#password').value;

		setErrorState('');

		if (email === '' || password === '') {
			console.error('Invalid email and password');
			setErrorState('Invalid email and password');
			return;
		}

		const resp = await fetch(`${URL}/signin`, {
			method: 'POST',
			headers: { 'Content-Type': 'application/json' },
			body: JSON.stringify({
				email,
				password
			})
		});

		const activeUser = await resp.json();

		if (resp.status === 200) {
			setUser(activeUser);
			setRoute('login');
		} else {
			console.error('Invalid email and password');
			setErrorState('Invalid email and password');
		}
	};

	const handleEnterKey = event => {
		if (event.key === 'Enter') {
			handleSubmit();
		}
	};

	return (
		<div>
			<article
				id='form-container'
				className='mw6 center bg-white-20 br3 pa3 pa3-ns mv3 ba b--black-10 shadow-5'
			>
				<main className='pa4 black-80'>
					<div className='measure'>
						<fieldset id='sign_up' className='ba b--transparent ph0 mh0'>
							<legend className='f2 fw6 ph0 mh0'>Sign In</legend>
							<div className='mt3'>
								<label className='db fw6 lh-copy f6' htmlFor='email-address'>
									Email
								</label>
								<input
									className='pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100'
									type='email'
									name='email-address'
									id='email-address'
									onKeyPress={handleEnterKey}
								/>
							</div>
							<div className='mv3'>
								<label className='db fw6 lh-copy f6' htmlFor='password'>
									Password
								</label>
								<input
									className='b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100'
									type='password'
									name='password'
									id='password'
									onKeyPress={handleEnterKey}
								/>
							</div>
							<div className='mt4'>
								<input
									className='b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib'
									type='submit'
									value='Sign in'
									onClick={handleSubmit}
								/>
							</div>
							<div className='lh-copy mt3'>
								<p
									className='f5 link black db bolder mb1 pointer'
									onClick={() => setRoute(route)}
								>
									Register
								</p>
							</div>
						</fieldset>
					</div>
				</main>
			</article>
			{errorState ? (
				<div
					id='error-message'
					className='f4 mt4 dark-red animate__animated animate__heartBeat'
				>
					{errorState}
				</div>
			) : null}
		</div>
	);
};

const mapStateToProps = state => ({
	route: state.setRoute.route
});

const mapDispatchToProps = dispatch => ({
	setRoute: route => dispatch(updateRoute(route)),
	setUser: user => dispatch(setUserProfile(user))
});

export default connect(mapStateToProps, mapDispatchToProps)(SignIn);
