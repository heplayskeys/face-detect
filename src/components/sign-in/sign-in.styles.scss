$breakpoint-phone: 450px;

.bolder {
	&:hover {
		font-weight: bold;
		color: #000;
	}
}

#form-container {
	@media screen and (max-width: $breakpoint-phone) {
		margin-top: 10px;
		border: none;
		background: none;
		box-shadow: none;
	}
}

#error-message {
	@media screen and (max-width: $breakpoint-phone) {
		margin: 0;
	}
}
