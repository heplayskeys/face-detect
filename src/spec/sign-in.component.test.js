import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { SignIn } from '../components/sign-in/sign-in.component';

let wrapper;

describe('<SignIn />', () => {
	beforeAll(() => {
		const mockProps = {
			URL: '',
			route: '',
			setRoute: jest.fn(),
			setUser: jest.fn()
		};
		wrapper = shallow(<SignIn {...mockProps} />);
	});

	it('renders register component', () => {
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});
});
