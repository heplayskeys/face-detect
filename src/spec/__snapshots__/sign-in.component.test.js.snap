// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> renders register component 1`] = `
<div>
  <article
    className="mw6 center bg-white-20 br3 pa3 pa3-ns mv3 ba b--black-10 shadow-5"
    id="form-container"
  >
    <main
      className="pa4 black-80"
    >
      <div
        className="measure"
      >
        <fieldset
          className="ba b--transparent ph0 mh0"
          id="sign_up"
        >
          <legend
            className="f2 fw6 ph0 mh0"
          >
            Sign In
          </legend>
          <div
            className="mt3"
          >
            <label
              className="db fw6 lh-copy f6"
              htmlFor="email-address"
            >
              Email
            </label>
            <input
              className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
              id="email-address"
              name="email-address"
              onKeyPress={[Function]}
              type="email"
            />
          </div>
          <div
            className="mv3"
          >
            <label
              className="db fw6 lh-copy f6"
              htmlFor="password"
            >
              Password
            </label>
            <input
              className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
              id="password"
              name="password"
              onKeyPress={[Function]}
              type="password"
            />
          </div>
          <div
            className="mt4"
          >
            <input
              className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib"
              onClick={[Function]}
              type="submit"
              value="Sign in"
            />
          </div>
          <div
            className="lh-copy mt3"
          >
            <p
              className="f5 link black db bolder mb1 pointer"
              onClick={[Function]}
            >
              Register
            </p>
          </div>
        </fieldset>
      </div>
    </main>
  </article>
</div>
`;
